test/test_infra/helpers/gc.dart

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:developer';
+
+/// Forces garbage collection by aggressive memory allocation.
+Future<void> forceGC() async {
+  final _storage = <List<DateTime>>[];
+
+  void allocateMemory() {
+    _storage.add(Iterable.generate(10000, (_) => DateTime.now()).toList());
```

Comment:
If we are running these stress tests on our github bots, we may need to double check if we will be taxing any shared quotas.
Do you know if this is something that will take a lot of resources on the bots, or use a lot of time units?

Reply:
If everything is as expected, this should not take a lot of resources: the cycle will stop at first garbage collection, that should happen much earlier than any significant impact for bots.